internal/resource/config/oauth/issuer/oauth_issuer_data_source.go

```diff
@@ -0,0 +1,123 @@
+var (
+	_ datasource.DataSource              = &oauthIssuerDataSource{}
+	_ datasource.DataSourceWithConfigure = &oauthIssuerDataSource{}
+)
+
+// Create a Administrative Account data source
+func NewOauthIssuerDataSource() datasource.DataSource {
+	return &oauthIssuerDataSource{}
+}
+
+// oauthIssuerDataSource is the datasource implementation.
+type oauthIssuerDataSource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+type oauthIssuerDataSourceModel struct {
+	Id          types.String `tfsdk:"id"`
+	CustomId    types.String `tfsdk:"custom_id"`
+	Name        types.String `tfsdk:"name"`
+	Description types.String `tfsdk:"description"`
+	Host        types.String `tfsdk:"host"`
+	Path        types.String `tfsdk:"path"`
+}
+
+// GetSchema defines the schema for the datasource.
+func (r *oauthIssuerDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	schemaDef := schema.Schema{
+		Description: "Describes an OAuth Issuer.",
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Required: false,
+				Optional: false,
+				Computed: true,
+			},
+			"description": schema.StringAttribute{
+				Required: false,
+				Optional: false,
+				Computed: true,
+			},
+			"host": schema.StringAttribute{
+				Required: false,
+				Optional: false,
+				Computed: true,
+			},
+			"path": schema.StringAttribute{
+				Required: false,
+				Optional: false,
+				Computed: true,
+			},
+		},
+	}
+	id.ToDataSourceSchema(&schemaDef, false, "The persistent, unique ID for the virtual issuer. It can be any combination of [a-zA-Z0-9._-]. This property is system-assigned if not specified.")
+	id.ToDataSourceSchemaCustomId(&schemaDef, true, true, "The persistent, unique ID for the virtual issuer. It can be any combination of [a-zA-Z0-9._-]. This property is system-assigned if not specified.")
+	resp.Schema = schemaDef
+}
+
+// Metadata returns the data source type name.
+func (r *oauthIssuerDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_oauth_issuer"
+}
+
+// Configure adds the provider configured client to the data source.
+func (r *oauthIssuerDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, _ *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+}
+
+// Read a OauthIssuerResponse object into the model struct
+func readOauthIssuerResponseDataSource(ctx context.Context, r *client.Issuer, state *oauthIssuerDataSourceModel) {
+	//TODO why is this a pointer?
+	state.Id = types.StringValue(*r.Id)
+	state.CustomId = types.StringValue(*r.Id)
+	state.Name = types.StringValue(r.Name)
+	state.Description = types.StringValue(*r.Description)
```

Comment:
Use StringPointerValue. Same for state.Path below.

Reply:
StringPointerValue's are now used